lldb/include/lldb/Utility/StructuredData.h

```diff
@@ -290,6 +290,36 @@ class StructuredData {
 
     void GetDescription(lldb_private::Stream &s) const override;
 
+    /// Creates an Array of substrings by splitting a string around the
+    /// occurrences of a separator character.
+    ///
+    /// Note:
+    /// * This is almost the same API and implementation as `StringRef::split`.
+    /// * Not depending on `StringRef::split` because it will involve a
+    ///   temporary `SmallVectorImpl`.
```

Comment:
It sounds like this is justifying duplication for efficiency reasons. Have you actually observed that there is an actual gain in duplicating this functionality here? If you're doing this preemptively because it sounds like it should be faster, I would say measure or don't do it at all.

Reply:
Updated the implementation to call `StringRef::split` with a `SmallVector`, then move the substrings into the to-be-returned array.